database/src/test/kotlin/fi/vauhtijuoksu/vauhtijuoksuapi/database/impl/GameDataDatabaseTest.kt

```diff
@@ -43,4 +48,54 @@ class GameDataDatabaseTest : VauhtijuoksuDatabaseTest<GameData>() {
     override fun getDatabase(injector: Injector): VauhtijuoksuDatabase<GameData> {
         return injector.getInstance(GameDataDatabase::class.java)
     }
+
+    @Test
+    fun testUpdate(testContext: VertxTestContext) {
+        val oldId = TestGameData.gameData1.id
+        val newGame = TestGameData.gameData2.copy(id = oldId)
+        db.update(TestGameData.gameData2.copy(id = oldId))
+            .onFailure(testContext::failNow)
+            .onSuccess { res ->
+                testContext.verify {
+                    assertEquals(res, newGame)
+                }
+            }.compose {
+                db.getById(oldId!!)
+            }
+            .onFailure(testContext::failNow)
+            .onSuccess { res ->
+                testContext.verify {
+                    assertEquals(res, newGame)
+                }
+            }.compose {
```

Comment:
> I don't think you can compose after success
You did so here. On a second look, it you'd need to return the id from previous step, but that shouldn't be a problem either

Reply:
val != var oli ongelma